src/multicast.ts

```diff
@@ -0,0 +1,42 @@
+import { Producer, StrictSink } from './types';
+
+export function multicast<T>(source: Producer<T>): Producer<T> {
+  let sinks: StrictSink<T>[] = [];
+  let talkback: any;
+  let lasts: T[] = [];
+  let timerActive = false;
+
+  source(0, (t, d) => {
```

Comment:
The PR looks all good, except this line here made me think a bit. Because we're calling `source(0,...` here at the "setup" phase of the operator, it's subscribing immediately to the source, before the returned multicasted callbag has been subscribed to. In other words, in RxJS terminology, this is like `source.multicast(new Subject()).connect()`, with an emphasis on `connect()`.

My opinion here has two suggestions: (1) I don't actually mind that this operator works like this **as long as** we use this only in Cycle.js internals, and as long as we remind ourselves of this subscribe-ASAP behavior, and preferably if we make that behavior explicit in the name, e.g. `multicastNow` or `subscribeAndMulticast`. The name similarity with RxJS `multicast` could come back to bite us. If `cyclejs/callbags` gets used by anyone else than us for the internals of Cycle.js, they will also be bitten by this behavior.

The option is (2) implement `multicast` with refCount built-in instead, which means that in line 30 we add a check if the pushed sink is the first one (then subscribe and multicast the source), and when the last popped (actually `filter`ed) sink leaves, then unsubscribe from the multicasted source.

Reply:
> Why is multicast so weird? Now I get various bugs related to missing termination

Maybe we're just missing refCounting?